odoo/http.py

```diff
@@ -1144,17 +1152,46 @@ def make_response(self, data, headers=None, cookies=None):
         complete response object, or the returned data will not be correctly
         interpreted by the clients.
 
-        :param basestring data: response body
+        :param str data: response body
+        :param int status: http status code
         :param headers: HTTP headers to set on the response
         :type headers: ``[(name, value)]``
         :param collections.Mapping cookies: cookies to set on the client
+        :returns: a response object.
+        :rtype: :class:`~odoo.http.Response`
         """
-        response = Response(data, headers=headers)
+        response = Response(data, status=status, headers=headers)
         if cookies:
             for k, v in cookies.items():
                 response.set_cookie(k, v)
         return response
 
+    def make_json_response(self, data, headers=None, cookies=None, status=200):
+        """ Helper for JSON responses, it json-serializes ``data`` and
+        sets the Content-Type header accordingly.
+
+        :param data: the data that will be serialized into the response body
+        :type data: anything that is json-serializable
+        :param int status: http status code
+        :param headers: HTTP headers to set on the response
+        :type headers: ``[(name, value)]``
+        :param collections.Mapping cookies: cookies to set on the client
+        :returns: a response object.
+        :rtype: :class:`~odoo.http.Response`
+            """
+        if headers is None:
+            headers = []
+
+        data = json.dumps(data, ensure_ascii=False)
+        for header_name, header_value in headers:
+            if header_name.lower() == 'content-type':
+                assert 'charset=utf8' in header_value.lower().replace('-', '')
```

Comment:
I don't think that's useful, or even correct: for starters leaving the charset out is perfectly normal.